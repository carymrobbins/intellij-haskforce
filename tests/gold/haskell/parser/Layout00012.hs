module Layout00012 where

type BoundedInt a = (BoundedSuper a, BoundedSuper (UnsignedRep a))

-- | Super class to 'BoundedInt'
#if defined(__GLASGOW_HASKELL__) && __GLASGOW_HASKELL__ >= 708
class    (Ord a, Num a, Bounded a, Integral a, FiniteBits a) => BoundedSuper a
#endif
