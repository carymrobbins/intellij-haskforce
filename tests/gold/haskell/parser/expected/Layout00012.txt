Haskell File
  MODULEDECL
    PsiElement(HaskellTokenType.module)('module')
    QCONID
      CONID
        PsiElement(HaskellTokenType.conidRegexp)('Layout00012')
    PsiElement(HaskellTokenType.where)('where')
  BODY
    TYPEDECL
      PsiElement(HaskellTokenType.type)('type')
      TYPEE
        ATYPE
          OQTYCON
            QTYCON
              TYCON
                CONID
                  PsiElement(HaskellTokenType.conidRegexp)('BoundedInt')
        ATYPE
          TYVAR
            VARID
              PsiElement(HaskellTokenType.varidRegexp)('a')
      PsiElement(HaskellTokenType.=)('=')
      TYPEE
        ATYPE
          PsiElement(HaskellTokenType.()('(')
          CTYPE
            TYPEE
              ATYPE
                OQTYCON
                  QTYCON
                    TYCON
                      CONID
                        PsiElement(HaskellTokenType.conidRegexp)('BoundedSuper')
              ATYPE
                TYVAR
                  VARID
                    PsiElement(HaskellTokenType.varidRegexp)('a')
          PsiElement(HaskellTokenType.,)(',')
          CTYPE
            TYPEE
              ATYPE
                OQTYCON
                  QTYCON
                    TYCON
                      CONID
                        PsiElement(HaskellTokenType.conidRegexp)('BoundedSuper')
              ATYPE
                PsiElement(HaskellTokenType.()('(')
                CTYPE
                  TYPEE
                    ATYPE
                      OQTYCON
                        QTYCON
                          TYCON
                            CONID
                              PsiElement(HaskellTokenType.conidRegexp)('UnsignedRep')
                    ATYPE
                      TYVAR
                        VARID
                          PsiElement(HaskellTokenType.varidRegexp)('a')
                PsiElement(HaskellTokenType.))(')')
          PsiElement(HaskellTokenType.))(')')
    PsiComment(HaskellTokenType.haddock)('-- | Super class to 'BoundedInt'\n')
    PsiComment(HaskellTokenType.CPPIF)('#if defined(__GLASGOW_HASKELL__) && __GLASGOW_HASKELL__ >= 708')
    CLASSDECL
      PsiElement(HaskellTokenType.class)('class')
      CTYPE
        CLSCONTEXT
          ATYPE
            PsiElement(HaskellTokenType.()('(')
            CTYPE
              TYPEE
                ATYPE
                  OQTYCON
                    QTYCON
                      TYCON
                        CONID
                          PsiElement(HaskellTokenType.conidRegexp)('Ord')
                ATYPE
                  TYVAR
                    VARID
                      PsiElement(HaskellTokenType.varidRegexp)('a')
            PsiElement(HaskellTokenType.,)(',')
            CTYPE
              TYPEE
                ATYPE
                  OQTYCON
                    QTYCON
                      TYCON
                        CONID
                          PsiElement(HaskellTokenType.conidRegexp)('Num')
                ATYPE
                  TYVAR
                    VARID
                      PsiElement(HaskellTokenType.varidRegexp)('a')
            PsiElement(HaskellTokenType.,)(',')
            CTYPE
              TYPEE
                ATYPE
                  OQTYCON
                    QTYCON
                      TYCON
                        CONID
                          PsiElement(HaskellTokenType.conidRegexp)('Bounded')
                ATYPE
                  TYVAR
                    VARID
                      PsiElement(HaskellTokenType.varidRegexp)('a')
            PsiElement(HaskellTokenType.,)(',')
            CTYPE
              TYPEE
                ATYPE
                  OQTYCON
                    QTYCON
                      TYCON
                        CONID
                          PsiElement(HaskellTokenType.conidRegexp)('Integral')
                ATYPE
                  TYVAR
                    VARID
                      PsiElement(HaskellTokenType.varidRegexp)('a')
            PsiElement(HaskellTokenType.,)(',')
            CTYPE
              TYPEE
                ATYPE
                  OQTYCON
                    QTYCON
                      TYCON
                        CONID
                          PsiElement(HaskellTokenType.conidRegexp)('FiniteBits')
                ATYPE
                  TYVAR
                    VARID
                      PsiElement(HaskellTokenType.varidRegexp)('a')
            PsiElement(HaskellTokenType.))(')')
        PsiElement(HaskellTokenType.=>)('=>')
        CTYPE
          TYPEE
            ATYPE
              OQTYCON
                QTYCON
                  TYCON
                    CONID
                      PsiElement(HaskellTokenType.conidRegexp)('BoundedSuper')
            ATYPE
              TYVAR
                VARID
                  PsiElement(HaskellTokenType.varidRegexp)('a')
    PsiComment(HaskellTokenType.#endif)('#endif')
